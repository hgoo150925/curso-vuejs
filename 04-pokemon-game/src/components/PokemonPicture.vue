<template>
  <div class="pokemon-container">
    <img :src="imgSrc" class="hidden-pokemon" alt="pokemon" />
    <!-- si showPokemon es true se muestra la img -->
    <!-- showPokemon es una props -->
    <img v-if="showPokemon" :src="imgSrc" class="fade-in" alt="pokemon" />
  </div>
</template>

<script>
export default {
  props: {
    pokemonId: {
      type: Number,
      required: true,
    },
    showPokemon: {
      type: Boolean,
      required: true,
      default: false,
    },
  },

  // Las propiedades computadas se almacenan en caché según sus dependencias.
  // Una propiedad computada solo se volverá a evaluar cuando alguna de sus dependencias haya cambiado
  computed: {
    imgSrc() {
      return `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${this.pokemonId}.svg`;
    },
  },
};
</script>

<style scoped>
.pokemon-container {
  height: 200px;
}
img {
  height: 200px;
  position: absolute;
  width: 100vw;
  margin-left: -50vw;
  user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-drag: none;
  -webkit-user-select: none;
}
.hidden-pokemon {
  filter: brightness(0);
}
</style>
