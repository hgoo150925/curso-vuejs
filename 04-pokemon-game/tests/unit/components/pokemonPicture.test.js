import { shallowMount } from '@vue/test-utils';

import { describe, test, expect } from 'vitest';

import PokemonPicture from '../../../src/components/PokemonPicture.vue';

describe('PokemonPicture component', () => {
  test('debe de hacer match con el snapshot', () => {
    // Pasamos las props del componente
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        pokemonId: 1,
        showPokemon: false,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('debe de mostrar la imagen oculta y el pokemon 100', () => {
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        pokemonId: 100,
        showPokemon: false,
      },
    });

    const [img1, img2] = wrapper.findAll('img');

    expect(img1.exists()).toBeTruthy();
    expect(img2).toBe(undefined);

    // evaluar que la clase hidden-pokemon exista
    expect(img1.classes('hidden-pokemon')).toBe(true);
    expect(img1.attributes('src')).toBe(
      'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg'
    );
  });

  test('debe de mostrar el pokemon si showPokemon:true', () => {
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        pokemonId: 100,
        showPokemon: true,
      },
    });

    const img1 = wrapper.find('img');

    expect(img1.exists()).toBeTruthy();

    expect(img1.classes('hidden-pokemon')).toBe(false);
    expect(img1.classes('fade-in')).toBe(true);
  });
});
