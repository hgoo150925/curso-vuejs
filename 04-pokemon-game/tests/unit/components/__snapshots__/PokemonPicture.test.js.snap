// Vitest Snapshot v1

exports[`PokemonPicture component > debe de hacer match con el snapshot 1`] = `
"<div class=\\"pokemon-container\\" data-v-c76ad58e=\\"\\"><img src=\\"https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg\\" class=\\"hidden-pokemon\\" alt=\\"pokemon\\" data-v-c76ad58e=\\"\\"><!-- si showPokemon es true se muestra la img -->
  <!-- showPokemon es una props -->
  <!--v-if-->
</div>"
`;
